orttraining/orttraining/core/graph/gradient_builder_base.cc

```diff
@@ -405,5 +405,22 @@ AttributeProto GradientBuilderBase::AttributeDefinitionToAttributeProto(
   return attr_proto;
 }
 
+// NOTE THAT we are changing the ATenOp forward node. If any of the input requires grad, we need to:
+// 1. set the requires_grad attribute in ATenOp so we can get this info during ATenOp forward execution and pass to
+// ATenOp executor.
+// 2. Append a new CPU INT scalar to outputs to pass context ID to ATenOpGrad for backward execution.
+void GradientBuilderBase::HandleATenOpGradient(const std::vector<int64_t>& requires_grad) const {
```

Comment:
I feel  HandleATenOpGradient should be a private function for GetATenOpGradient class. 

we might skip using DECLARE_GRADIENT_BUILDER macro, and use an explicit GetATenOpGradient  class declaration?

Reply:
The problem is we are using graph_ and node_ inside this function, and they are private in GradientBuilderBase. We need to move them from private to protected if we want to do this.